theseus/optimizer/nonlinear/gauss_newton.py

```diff
@@ -41,3 +41,29 @@ def __init__(
 
     def compute_delta(self, **kwargs) -> torch.Tensor:
         return self.linear_solver.solve()
+
+    def compute_samples(self, n_samples: int = 10, T: float = 1.0) -> torch.Tensor:
```

Comment:
I am guessing `T` is temperature? Maybe pick a more descriptive name or describe in comments. Same for `B`, `N`, `R` below, where you can point to the equations from any reference.